### PR TITLE
Fetch sendgrid marketing lists

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This creates a helper to fetch sendgrid marketing lists and does some refactoring to simplify interactions with the sendgrid API.

Fixes SC-13280

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible.

### Author checklist

- [ ] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/utils/sendgrid/api.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	}
+
+	// Create the GET request
+	req := sendgrid.GetRequest(apiKey, Lists, Host)
+	req.Method = http.MethodGet
+	req.QueryParams = params
+
+	return doRequest(req)
+}
+
```
Comment (bbengfort) [inline pkg/utils/sendgrid/api.go:59]: Could we also add the field definitions request here as well?

```go
const sgFields   = "/v3/marketing/field_definitions"

func FieldDefinitions(apiKey string) (_ string, err error) {
	// Create the SendGrid request
	req := sendgrid.GetRequest(apiKey, sgFields, sgHost)
	req.Method = http.MethodGet

	rep, err := sendgrid.API(req)
	if err != nil {
		return "", fmt.Errorf("could not make sendgrid request: %w", err)
	}

	if rep.StatusCode < 200 || rep.StatusCode >= 300 {
		return "", fmt.Errorf("received status code %d: could not make sendgrid request", rep.StatusCode)
	}

	return rep.Body, nil
}
```

File: pkg/utils/sendgrid/api.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	"net/http"
+
+	"github.com/sendgrid/rest"
+	"github.com/sendgrid/sendgrid-go"
+)
+
+const (
+	Host     = "https://api.sendgrid.com"
+	Contacts = "/v3/marketing/contacts"
+	Lists    = "/v3/marketing/lists"
```
Comment (bbengfort) [inline pkg/utils/sendgrid/api.go:16]: Could we suffix these with an `EP` so it's more clear in the code that it's an endpoint or URL?

**Review by bbengfort — APPROVED** [review]
Quick and easy! Is there a follow on task to create a CLI command that can fetch the List and Field IDs?

File: pkg/utils/sendgrid/api.go (hunk 1 above)
Comment (pdeziel) [reply pkg/utils/sendgrid/api.go:59]: Will do!

**Pull request merged**